pkg/transaction/eventgenerator.go

```diff
@@ -59,19 +61,127 @@ func (e *Generator) SetUseTrafficForAggregation(useTrafficForAggregation bool) {
 
 // CreateEvent - Creates a new event to be sent to Amplify Observability, expects sampling is handled by agent
 func (e *Generator) CreateEvent(logEvent LogEvent, eventTime time.Time, metaData common.MapStr, eventFields common.MapStr, privateData interface{}) (beat.Event, error) {
-	// if CreateEvent is being used, sampling will not work, so all events need to be sent
-	if metaData == nil {
-		metaData = common.MapStr{}
+	builder := NewEventReportBuilder().
+		SetEventTime(eventTime).
+		SetMetadata(metaData).
+		SetFields(eventFields).
+		SetPrivateData(privateData).
+		SetForceSample()
+
+	// set the proper log event type
+	if logEvent.TransactionSummary != nil {
+		builder = builder.SetSummaryEvent(logEvent)
+	} else {
+		builder = builder.SetDetailEvents([]LogEvent{logEvent}).SetSkipMetricTracking()
 	}
-	metaData.Put(sampling.SampleKey, true)
 
-	if logEvent.TransactionSummary != nil {
+	report, err := builder.Build()
+	if err != nil {
+		return beat.Event{}, err
+	}
+
+	events, err := e.CreateFromEventReport(report)
+	if err != nil {
+		return beat.Event{}, err
+	}
+	if len(events) == 0 {
+		return beat.Event{}, errors.New("an event was not created")
+	}
+	if len(events) > 1 {
+		return events[0], errors.New("unexpectedly, more than one event was created, only returning the first")
```

Comment:
since we are getting only the first one, regardless of if theres more than 1, should this just warn in the if statement, and then fall through to the final return?

Reply:
...unless you really really wanted to return an error?

Reply:
that would be the agent implementations issue here